taf/GitRepository.py

```diff
@@ -20,32 +22,85 @@ def __init__(self, root_dir, target_path=None, repo_urls=None, additional_info=N
     root_dir.
     """
     self.target_path = target_path
-    self.repo_path = str((Path(root_dir) / (target_path or '')).resolve(True))
+    self.repo_path = str((Path(root_dir) / (target_path or '')).resolve())
     self.repo_urls = repo_urls
     self.additional_info = additional_info
+    self.bare = bare
+
+  @property
+  def is_git_repository(self):
+    return self._is_git_repository()
 
   @property
   def name(self):
     return os.path.basename(self.repo_path)
 
-  def _git(self, cmd):
-    return run(f'git -C {self.repo_path} {cmd}')
+  def _git(self, cmd, *args):
+    """Call git commands in subprocess
+
+    E.G.:
+      self._git('checkout {}', branch_name)
+    """
+    if len(args):
+      cmd = cmd.format(*args)
+    return run('git -C {} {}'.format(self.repo_path, cmd))
+
+  def all_commits_since_commit(self, since_commit):
+    if since_commit is not None:
+      commits = self._git('rev-list {}..HEAD', since_commit).strip()
+    else:
+      commits = self._git('log --format=format:%H').strip()
+    if not commits:
+      return []
+    commits = commits.split('\n')
+    commits.reverse()
+    return commits
+
+  def all_fetched_commits(self, branch='master'):
+    commits = self._git('rev-list ..origin/{}'.format(branch)).strip()
```

Comment:
We can now call `_git` method this way: `self._git('rev-list ..origin/{}', branch).strip()`.